src/hotspot/share/runtime/thread.cpp

```diff
@@ -2205,22 +2214,41 @@ const char* JavaThread::name() const  {
 // descriptive string if there is no set name.
 const char* JavaThread::get_thread_name_string(char* buf, int buflen) const {
   const char* name_str;
-  oop thread_obj = threadObj();
-  if (thread_obj != NULL) {
-    oop name = java_lang_Thread::name(thread_obj);
-    if (name != NULL) {
-      if (buf == NULL) {
-        name_str = java_lang_String::as_utf8_string(name);
+  Thread* current = Thread::current_or_null();
+  if (current == nullptr) {
+    // Current thread is not attached so it can't safely determine this
+    // JavaThread's name so use the default thread name.
```

Comment:
This should not be possible, but again `current_or_null_safe` should be used.

Though this code is used a lot in normal operation so the additional overhead of this is more significant than the print_on_error case.

I think this check should be moved to the caller if  needed (ie the print_on_error code), as in normal use it is not possible to fail this check.

Reply:
I'm going to switch back to `Thread::current()` and remove the extra handling for
the `current == nullptr` case.